### PR TITLE
Add screen capture functionality for TekScope device driver family

### PR DESCRIPTION
**Pull request by nfelt14**
## Proposed changes

feat: Added a screen capture mixin and implemented the functionality for the TekScope family of device drivers

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [x] New feature (non-breaking change which adds functionality)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have followed the guidelines in the [CONTRIBUTING](https://github.com/tektronix/tm_devices/blob/main/CONTRIBUTING.md) document
- [x] I have signed the CLA
- [x] I have checked to ensure there aren't other open [Pull Requests](https://github.com/tektronix/tm_devices/pulls) for the same update/change
- [x] I have created (or updated) an [Issue](https://github.com/tektronix/tm_devices/issues) to track the status of this update/change and updated the link in this PR description (see above in the **Proposed changes** section) using the wording `Addresses #<issue_number>`
- [x] I have performed a self-review of my code
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] Basic linting passes locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have added necessary documentation (if appropriate)
- [x] I have updated the Changelog with a brief description of my changes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -20,6 +20,8 @@ Things to be included in the next release go here.
 
 ### Added
 
+- Added a new mixin, `ScreenCaptureMixin`, that defines methods/properties used for capturing screenshots from devices.
+- Added screen capture capabilities to the `TekScope` family of device drivers.
```
Comment (nfelt14) [inline CHANGELOG.md:24]: Probably should add a new example to the documentation showing how to use this.

**Review by michaelwagoner — APPROVED** [review]

File: src/tm_devices/driver_mixins/abstract_device_functionality/screen_capture_mixin.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        local_folder: Union[str, os.PathLike[str]] = "./",
+        device_folder: Union[str, os.PathLike[str]] = "./",
+        keep_device_file: bool = False,
+    ) -> None:
+        """Capture a screenshot from the device and save it locally.
+
+        Args:
+            filename: The name of the file to save the screenshot as.
+            colors: The color scheme to use for the screenshot.
+            view_type: The type of view to capture.
```
Comment (ldantek) [inline src/tm_devices/driver_mixins/abstract_device_functionality/screen_capture_mixin.py:44]: I don't understand this arg description, and I don't see it used in the tests or examples. I think it needs more context. and if it's something specific to scopes, let's discuss how to handle that.

Comment (nfelt14) [reply src/tm_devices/driver_mixins/abstract_device_functionality/screen_capture_mixin.py]: I updated the docstrings

**Review by ldantek — DISMISSED** [review]

**Comment by amccann-Tek** [comment]
Couple questions: 
What are your thoughts on not requiring a file name and just using the standard date time format?
And
Assuming we can trigger and take multiple events a second, are we able to increment or handle the case where there might be a duplicate file name collision?

**Comment by nfelt14** [comment]
> Couple questions: What are your thoughts on not requiring a file name and just using the standard date time format? And Assuming we can trigger and take multiple events a second, are we able to increment or handle the case where there might be a duplicate file name collision?

My thoughts would be to let the user decide if they want the filename to be datetime or not, then tm_devices doesn't need to worry about name collisions, since that is out of its control.

**Comment by amccann-Tek** [comment]
> > Couple questions: What are your thoughts on not requiring a file name and just using the standard date time format? And Assuming we can trigger and take multiple events a second, are we able to increment or handle the case where there might be a duplicate file name collision?
> 
> My thoughts would be to let the user decide if they want the filename to be datetime or not, then tm_devices doesn't need to worry about name collisions, since that is out of its control.

For that first part I meant more along the lines of allowing 
save_screenshot() to be valid, and default output would be date_time naming.

**Comment by nfelt14** [comment]
> > > Couple questions: What are your thoughts on not requiring a file name and just using the standard date time format? And Assuming we can trigger and take multiple events a second, are we able to increment or handle the case where there might be a duplicate file name collision?
> > 
> > My thoughts would be to let the user decide if they want the filename to be datetime or not, then tm_devices doesn't need to worry about name collisions, since that is out of its control.
> 
> For that first part I meant more along the lines of allowing 
> save_screenshot() to be valid, and default output would be date_time naming.

I could add that in.

**Review by amccann-Tek — APPROVED** [review]

File: examples/scopes/tekscope/save_screenshot.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+    # Save a screenshot as example.png, will create a screenshot on the device,
+    # copy it to the current working directory on the local machine,
+    # and then delete the screenshot file from the device.
+    scope.save_screenshot("example.png")
+
+    # Save a screenshot as example.jpg, will create a screenshot on the device,
+    # copy it to ./images/example.jpg on the local machine,
+    # but this time the screenshot file on the device will not be deleted.
+    scope.save_screenshot("example.jpg", local_folder="./images", keep_device_file=True)
```
[the remark this answers is not in the document]
Comment (nfelt14) [reply examples/scopes/tekscope/save_screenshot.py:28]: I will add that

**Review by ldantek — APPROVED** [review]

**Pull request merged**